### ISSUE TITLE
[Bug] NotSupportedException when referencing Microsoft.SemanticKernel library v1.6.1

### ISSUE THREAD

**Issue by marcominerva** [microsoft/kernel-memory#361]
### Context / Scenario

Kernel Memory (in particular the **AzureOpenAI** project) has a dependency on **Microsoft.SemanticKernel.Connectors.OpenAI v1.4**.

If we add the **KernelMemory.Core** NuGet Package to a project, and then explicitly add a reference to **Microsoft.SemanticKernel v1.6.1**, the `KernelMemoryBuilder.Build` methods fails with a `NotSupportedException `exception

### What happened?

Digging in the source code, the problem is caused, for example, in the `AzureOpenAITextGenerator` class, in particular when it instantiate the `OpenAIClientOptions` object:

https://github.com/microsoft/kernel-memory/blob/8aa175443d24d212d42189a16e1bc832e6800a40/extensions/AzureOpenAI/AzureOpenAITextGenerator.cs#L69-L77

The constructor of this class, that is contained in the **Azure.AI.OpenAI** library, uses an Enum to determine with API version to use. The verison used by **KernelMemory.AzureOpenAI** (beta13) causes a mismatch with the version used by **Microsoft.SemanticKernel v1.6.1** (beta14), resulting in a `NotSupportedException`.

### Importance

I cannot use Kernel Memory

### Platform, Language, Versions

KernelMemory v 0.33.240312.1
SemanticKernel v1.6.1

**Comment by Ivaznzhou** [comment]
Reflector Microsoft.KernelMemory.AI.OpenAI, Version=0.33.0.0, Culture=neutral, PublicKeyToken=f300afd708cefcd3
Code 
OpenAIClientOptions options1 = new OpenAIClientOptions(OpenAIClientOptions.ServiceVersion.V2024_02_15_Preview | OpenAIClientOptions.ServiceVersion.V2023_05_15);

OpenAIClientOptions.ServiceVersion.V2024_02_15_Preview | OpenAIClientOptions.ServiceVersion.V2023_05_15 =7
But 
Azure.AI.OpenAI OpenAIClientOptions.ServiceVersion no 7

**Comment by Ivaznzhou** [comment]
screenshot
![image](https://github.com/microsoft/kernel-memory/assets/83056816/16f31618-b32a-4d44-b4e3-0c5d3bd02a22)

**Comment by dluc** [comment]
when calling `OpenAIClientOptions options = new()` we don't pass a version, leaving it to the class to use the default value. Did they remove the default value in beta14?

**Comment by dluc** [comment]
@markwallace-microsoft I might be wrong, but this looks like a breaking change, either in SK or indirectly via the adoption of Azure.AI beta14

**Comment by marcominerva** [comment]
The enum items have changed from beta13 to beta14, so now there is a mismatch:

Latest version:
https://github.com/Azure/azure-sdk-for-net/blob/db032227b881192bd22da59c37a45bdf43cd5a2e/sdk/openai/Azure.AI.OpenAI/src/Generated/OpenAIClientOptions.cs#L19-L31

```
public enum ServiceVersion
{
    /// <summary> Service version "2022-12-01". </summary>
    V2022_12_01 = 1,
    /// <summary> Service version "2023-05-15". </summary>
    V2023_05_15 = 2,
    /// <summary> Service version "2023-06-01-preview". </summary>
    V2023_06_01_Preview = 3,
    /// <summary> Service version "2023-07-01-preview". </summary>
    V2023_07_01_Preview = 4,
    /// <summary> Service version "2024-02-15-preview". </summary>
    V2024_02_15_Preview = 5,
}
```

Previous version:
https://github.com/Azure/azure-sdk-for-net/blob/a625fe176928ad8f62a99e4b5221e36608db49bb/sdk/openai/Azure.AI.OpenAI/src/Generated/OpenAIClientOptions.cs#L19-L35

```
public enum ServiceVersion
{
    /// <summary> Service version "2022-12-01". </summary>
    V2022_12_01 = 1,
    /// <summary> Service version "2023-05-15". </summary>
    V2023_05_15 = 2,
    /// <summary> Service version "2023-06-01-preview". </summary>
    V2023_06_01_Preview = 3,
    /// <summary> Service version "2023-07-01-preview". </summary>
    V2023_07_01_Preview = 4,
    /// <summary> Service version "2023-08-01-preview". </summary>
    V2023_08_01_Preview = 5,
    /// <summary> Service version "2023-09-01-preview". </summary>
    V2023_09_01_Preview = 6,
    /// <summary> Service version "2023-12-01-preview". </summary>
    V2023_12_01_Preview = 7,
}
```

**Comment by dluc** [comment]
thanks for the details, that's troubling, because it makes SK upgrade a breaking change in unrelated projects

**Comment by dluc** [comment]
Should be solved, #362 merged and released - thanks!!
